lua/lspconfig/server_configurations/golangci_lint_ls.lua

```diff
@@ -5,10 +5,10 @@ return {
     cmd = { 'golangci-lint-langserver' },
     filetypes = { 'go', 'gomod' },
     init_options = {
-      command = { 'golangci-lint', 'run', '--enable-all', '--disable', 'lll', '--out-format', 'json' },
+      command = { 'golangci-lint', 'run', '--out-format', 'json' },
     },
     root_dir = function(fname)
-      return util.root_pattern 'go.work'(fname) or util.root_pattern('go.mod', '.git')(fname)
+      return util.root_pattern('.golangci.yaml')(fname) or util.root_pattern('go.work')(fname) or util.root_pattern('go.mod', '.git')(fname)
```

Comment:
I'm not certain how to best use `util.root_pattern`.
Can't this line just be the following?

```lua
return util.root_pattern('.golangci.yaml', 'go.work', 'go.mod', '.git')(fname)
````

Reply:
The difference is, in the current case, `.golanci.yaml` is first searched for (up until the filesystem boundary), then `.go.work`, then `'go.mod` or `.git` (each level of the FHS is checked for either go.mod or .git`

In the latter case, all 4 files are checked for at each level of the hierarchy.

This is mostly done for projects with submodules/similar hierarchical structure. `go.work` tends to be the most unambiguous root marker.